### PR TITLE
Fix sonarqube suggested bugs

### PR DESCRIPTION
**Pull request by anandjakhaniya**
## Impact and Context
This PR only contains all `try-with-resource` changes.

**Before:**
This is from my local sonar setup.
image (3)

**After:**
image

## Testing
- [x] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

File: backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public abstract class Config {
   @JsonProperty private boolean event_system_enabled = false;
   @JsonProperty private ServerInterceptor tracingServerInterceptor = null;
   @JsonProperty private ClientInterceptor tracingClientInterceptor = null;
-  @JsonProperty private volatile OpenTelemetry openTelemetry;
+  @JsonProperty private OpenTelemetry openTelemetry;
```
Comment (jkwatson-verta) [inline backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java:39]: This field isn't actually used by anything, anywhere, as far as I can tell. We should remove it.

File: backend/server/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 2]
```diff
@@ -139,9 +142,6 @@ public static boolean needToRetry(Exception ex) {
   }
 
   public static Throwable findCommunicationsFailedCause(Throwable throwable) {
-    if (throwable == null) {
-      return null;
-    }
     var rootCause = throwable;
     while (rootCause.getCause() != null
         && !(rootCause.getCause() instanceof SocketException
```
Comment (jkwatson-verta) [inline backend/server/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java:147]: are we sure this null check isn't necessary? It looks like if we skip it, we'll get an NPE on line 146 if a null `throwable` is passed in.

**Review by jkwatson-verta — APPROVED** [review]
Just one question about a deleted null check.

File: backend/server/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/server/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java:147]: We have that check already but at the one step ahead where this function is called.

**Pull request merged**